iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2018,6 +2018,19 @@ public List<String> getPartitionNames(org.apache.hadoop.hive.ql.metadata.Table h
     return IcebergTableUtil.getPartitionNames(icebergTable, partitionSpec, true);
   }
 
+  /**
+   * Returns a list of partition names
+   * @param hmsTable A Hive table instance.
+   * @return A list of partition names for all partitions
+   */
+  @Override
+  public List<String> getPartitionNames(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
```

Comment:
can't we reuse the existing `getPartitionNames` and pass an empty spec?

Reply:
We can not use it, since it gives back partition names instead of partition keys, what needs here.
E.g. "event_src_trunc=BBB/event_time_month=2024-07" instead of "event_src".